crates/swc_ecma_parser/src/lexer/token.rs

```diff
@@ -0,0 +1,1493 @@
+            _ => return None,
+        };
+        Some(atom)
+    }
+
+    pub(crate) const fn is_keyword(self) -> bool {
+        let t = self as u8;
+        t >= Token::Await as u8 && t <= Token::Module as u8
+    }
+
+    pub(crate) fn as_known_ident_atom(&self) -> Option<Atom> {
+        let atom = match self {
+            Token::Abstract => atom!("abstract"),
+            Token::Any => atom!("any"),
+            Token::As => atom!("as"),
+            Token::Asserts => atom!("asserts"),
+            Token::Assert => atom!("assert"),
+            Token::Async => atom!("async"),
+            Token::Bigint => atom!("bigint"),
+            Token::Boolean => atom!("boolean"),
+            Token::Constructor => atom!("constructor"),
+            Token::Declare => atom!("declare"),
+            Token::Enum => atom!("enum"),
+            Token::From => atom!("from"),
+            Token::Get => atom!("get"),
+            Token::Global => atom!("global"),
+            Token::Implements => atom!("implements"),
+            Token::Interface => atom!("interface"),
+            Token::Intrinsic => atom!("intrinsic"),
+            Token::Is => atom!("is"),
+            Token::Keyof => atom!("keyof"),
+            Token::Namespace => atom!("namespace"),
+            Token::Never => atom!("never"),
+            Token::Number => atom!("number"),
+            Token::Object => atom!("object"),
+            Token::Of => atom!("of"),
+            Token::Package => atom!("package"),
+            Token::Private => atom!("private"),
+            Token::Protected => atom!("protected"),
+            Token::Public => atom!("public"),
+            Token::Readonly => atom!("readonly"),
+            Token::Require => atom!("require"),
+            Token::Set => atom!("set"),
+            Token::Static => atom!("static"),
+            Token::String => atom!("string"),
+            Token::Symbol => atom!("symbol"),
+            Token::Type => atom!("type"),
+            Token::Undefined => atom!("undefined"),
+            Token::Unique => atom!("unique"),
+            Token::Unknown => atom!("unknown"),
+            Token::Using => atom!("using"),
+            Token::Accessor => atom!("accessor"),
+            Token::Infer => atom!("infer"),
+            Token::Satisfies => atom!("satisfies"),
+            Token::Meta => atom!("meta"),
+            Token::Target => atom!("target"),
+            _ => return None,
+        };
+        Some(atom)
+    }
+
+    #[inline(always)]
+    pub(crate) const fn is_known_ident(self) -> bool {
+        let t = self as u8;
+        t >= Token::Abstract as u8 && t <= Token::Target as u8
+    }
+
+    pub(crate) const fn is_word(&self) -> bool {
+        matches!(
+            self,
+            Token::Null | Token::True | Token::False | Token::Ident
+        ) || (*self).is_known_ident()
+            || (*self).is_keyword()
+    }
+
+    pub(crate) fn as_word_atom(&self, value: Option<&TokenValue>) -> Option<Atom> {
```

Comment:
Same here, `Token` is `: Copy`. You don't need `&self` and needless referencing of Copy types end to make things slower